package.json

```
@@ -13,7 +13,9 @@
     "write-translations": "docusaurus write-translations",
     "write-heading-ids": "docusaurus write-heading-ids",
     "typecheck": "tsc",
-    "regenerate:local": "docs-sourcer"
+    "regenerate": "env-cmd docs-sourcer",
+    "regenerate:local": "yarn regenerate --local",
+    "regenerate:remote": "yarn regenerate --remote"
```

Won't these two lines need to update in accordance with the `docs-sourcer` parameter adjustment? Something like…
```suggestion
    "regenerate:local": "yarn regenerate --plugin local",
    "regenerate:remote": "yarn regenerate --all"
```

---

Or: maybe there's only…
```suggestion
    "regenerate": "env-cmd docs-sourcer --all",
    "regenerate:local": "env-cmd docs-sourcer --plugin local"
```

---

Forgive me, Ore, as I propose yet a third option. I'll add a little more context this time so you can follow my reasoning. While this all boils down to choosing the appropriate switches and defaults, that feels important to me because which we choose help to suggest the expected pattern of use, which makes it a UX concern.

In a standalone context, the tool is responsible for running a set of plugins in order to generate a docs site from various sources. The same can be said when running the tool in our cloud service — it's concerned with _everything_. For this reason, I think I may have misled in suggesting the need for a `--all` in the first place. That should probably be the default _for the `docs-sourcer` itself_.

In the context of the docs site, where we leverage yarn, it's _more_ likely that a user will want to do local-only regeneration, to view content they just edited. That's even more true in the long run, when the cloud process will manage changes to remote content automatically. There are still use cases for running remote generation in the docs repo manually, but they seem secondary.

For these reasons, I would propose that the `docs-sourcer` command runs all plugins by default, and accepts the `--plugin` and `--repo` arguments as a way to filter that to specific plugins. I'd then propose one of two options for the docs site itself

**OPTION 1:** Closest to standard usage, with just a shortcut for local generation as a convenience:
```
    "regenerate": "env-cmd docs-sourcer",
    "regenerate:local": "yarn regenerate --plugin local"
```

**OPTION 2:** Emphasizing the appropriate default for the context, at the risk of essentially reversing the default:
```
    "regenerate": "env-cmd docs-sourcer --plugin local",
    "regenerate:all": "env-cmd docs-sourcer"
```

Thoughts?

---

I am leaning with option 2 in the context of the docs repo. It is trivial to change so we test out any and switch if needed.

@eak12913 thoughts?

---

Upon reflection, I like option 2 best also. I think it reflects the pattern of use better, and no one but us will care that it follows a different pattern than the underlying `docs-sourcer` tool itself.